test/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter_test.cc

```diff
@@ -0,0 +1,107 @@
+#include "envoy/extensions/filters/network/sni_dynamic_forward_proxy/v3alpha/sni_dynamic_forward_proxy.pb.h"
+#include "envoy/network/connection.h"
+
+#include "extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.h"
+#include "extensions/filters/network/well_known_names.h"
+
+#include "test/extensions/common/dynamic_forward_proxy/mocks.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/mocks/upstream/transport_socket_match.h"
+
+using testing::AtLeast;
+using testing::Eq;
+using testing::InSequence;
+using testing::Return;
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace SniDynamicForwardProxy {
+namespace {
+
+using LoadDnsCacheEntryStatus = Common::DynamicForwardProxy::DnsCache::LoadDnsCacheEntryStatus;
+using MockLoadDnsCacheEntryResult =
+    Common::DynamicForwardProxy::MockDnsCache::MockLoadDnsCacheEntryResult;
+
+class ProxyFilterTest : public testing::Test,
+                        public Extensions::Common::DynamicForwardProxy::DnsCacheManagerFactory {
+public:
+  ProxyFilterTest() {
+    FilterConfig proto_config;
+    proto_config.set_port_value(443);
+    EXPECT_CALL(*dns_cache_manager_, getCache(_));
+    filter_config_ = std::make_shared<ProxyFilterConfig>(proto_config, *this, cm_);
+    filter_ = std::make_unique<ProxyFilter>(filter_config_);
+    filter_->initializeReadFilterCallbacks(callbacks_);
+
+    // Allow for an otherwise strict mock.
+    ON_CALL(callbacks_, connection()).WillByDefault(ReturnRef(connection_));
+    EXPECT_CALL(callbacks_, connection()).Times(AtLeast(0));
+
+    // Configure max pending to 1 so we can test circuit breaking.
+    cm_.thread_local_cluster_.cluster_.info_->resetResourceManager(0, 1, 0, 0, 0);
```

Comment:
I think you are missing a test for this circuit breaker?

Reply:
Circuit breaker is not implemented, added a TODO.